rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/PointerBlock.scala

```diff
@@ -0,0 +1,43 @@
+package coop.rchain.rspace.nextgenrspace.history
+import coop.rchain.rspace.Blake2b256Hash
+import coop.rchain.rspace.nextgenrspace.history.History.codecTrie
+import scodec.Codec
+import scodec.bits.BitVector
+import scodec.codecs.{provide, vectorOfN}
+import coop.rchain.shared.AttemptOps._
+
+final case class PointerBlock private (toVector: Vector[Trie]) {
+  def updated(tuples: List[(Int, Trie)]): PointerBlock =
+    new PointerBlock(tuples.foldLeft(toVector) { (vec, curr) =>
+      vec.updated(curr._1, curr._2)
```

Comment:
This should validate that the keys of each `tuple` are valid:
 - index in 0..255
 - value non-null (are there other exclusions?)

Reply:
I'm not convinced about it. we have another implementation of this PointerBlock (this is a shameless copy) which does not do any validation and there were no errors. my impression is that this is not a general purpose pointer block. it's a very particular piece of code that serves a particular purpose.

Reply:
GAH! I lost my comment...

Guess we're lucky to have this copy, now we can have a review for preexisting parts of the design as well :)

> there were no errors.

They might've slipped, they may still be lurking...

> not a general purpose pointer block

:100: and so it makes even more sense to encode usage-specific invariants in it, to catch bugs.

> a very particular piece of code that serves a particular purpose

This doesn't mandate assuming that only valid calls/usages of it are going to be made.

I'm fine with doing an `assert` if you want to avoid F[_]-ing with it ;)

Having said that all, I leave it up to you :)

Reply:
Completely disagree.
Every invariant, check, assert costs. We know how this class is being used. 
It's purpose is known. 
It will not change. 
Adding asserts, validations, checks only bloats a perfectly simple piece of code that's easy to comprehend and reason about.

Reply:
https://rchain.atlassian.net/browse/RCHAIN-3227